domain/lib/src/usecases/sort/sort_interactor.dart

```diff
@@ -38,23 +38,26 @@ class SortInteractor {
 
   SortInteractor(this._sortRepository);
 
-  Future<Either<Failure, Success>> execute<T>(List<T> listDocument, Sorter sorter) async {
-    final resultState = await catching(() => _sortRepository.sortFiles(listDocument, sorter))
-        .fold(
-            (exception) async {
-              switch(sorter.orderScreen) {
-                case OrderScreen.mySpace:
-                  return Left<Failure, Success>(MySpaceFailure(exception));
-              }
-            },
-            (documents) async {
-              final result = await documents;
-              if (result is List<Document>) {
-                return Right<Failure, Success>(MySpaceViewState(result as List<Document>));
-              } else {
-                return Right<Failure, Success>(MySpaceViewState([]));
-              }
-            });
-    return resultState;
+  Future<Either<Failure, Success>> execute<T>(List<T> listFiles, Sorter sorter) async {
+    try {
+      final filesSorted = await _sortRepository.sortFiles(listFiles, sorter);
+
+      if (filesSorted is List<Document>) {
+        return Right<Failure, Success>(MySpaceViewState(filesSorted as List<Document>));
```

Comment:
I don't think we need to use `as` here

Reply:
Because it is a generic type, the 'is' operator cannot be cast automatically. So I need to use the 'as' operator to cast it.

Reply:
We can use this way instead: `filesSorted.cast<Document>()`

Reply:
I think using as is safer to use cast. You can refer to this article [type-casting-collections-cast-as-retype-to-set-to-list](https://flutterbyexample.com/lesson/type-casting-collections-cast-as-retype-to-set-to-list)

Reply:
Oke i used it